onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_builder.cc

```diff
@@ -1859,8 +1878,16 @@ Status ConcatOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_builder, const
   const auto& input0 = inputs[0].node_arg.Name();
   const auto node_input_size = inputs.size();
 
-  // First if the inputs are uint8, we need verify all the inputs have same scale and zero points
-  if (operand_types.at(input0).type == android::nn::wrapper::Type::TENSOR_QUANT8_ASYMM) {
+  bool is_quant_op = IsQuantizedOp(node_unit);
+
+  // First if the inputs are uint8 and this is not a quantized Concat,
+  // we need verify all the inputs have same scale and zero points
+  // ONNX allows Concat (not QlinearConcat, not QDQ concat) run directly on uint8 without scales and zps,
+  // However, for NNAPI, all int8 inputs need scale and zero points cannot be 0
```

Comment:
Should this say 'all uint8 inputs'? TENSOR_QUANT8_ASYMM is unsigned isn't it?

Reply:
yes, this should be uint8